### PR TITLE
usb: samples: mass: add warning for no conf and nrf52840 conf

### LINKED ISSUE #14295: USB MSC sample has no default storage backend defined
**Describe the bug**
USB mass storage sample crashes during initialization. Device is not detected by the host.

**To Reproduce**
- `cd samples/subsys/usb/mass`
- `west build --board nrf52840_pca10056`
- `west flash`

**Expected behavior**
Proper initialization, a disk should be detected.

**Impact**
Showstopper.

**Screenshots or console output**
```
***** Booting Zephyr OS v1.14.0-rc1-1025-gc78c5e693679 *****                    
[00:00:00.004,272] <err> usb_msc: Storage init ERROR !!!! - Aborting USB init   
[00:00:00.004,272] <inf> main: The device is put in USB mass storage mode.   
```
**Environment (please complete the following information):**
 - OS: Ubuntu 16.04, 4.13.0
 - Toolchain: Zephyr SDK 0.9.5
 - Zephyr: c78c5e693679cc75f6e72e55c70f179ae9f6bea8

### PR DESCRIPTION
**Pull request by masz-nordic**
Added a warning that will abort build without disks.
Added a default configuration for nrf52840_pca10056.

Signed-off-by: Marcin Szymczyk <Marcin.Szymczyk@nordicsemi.no>

Fixes #14295

### CODE REVIEW

File: samples/subsys/usb/mass/CMakeLists.txt [hunk 1]
```diff
@@ -2,5 +2,9 @@ cmake_minimum_required(VERSION 3.13.1)
 include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
```
Comment (carlescufi) [inline samples/subsys/usb/mass/CMakeLists.txt]: You are missing this below before the `include` line:

```
if(BOARD STREQUAL nrf52840_pca10056)
  set(CONF_FILE nrf52840_pca10056.conf)
endif()
```

right?

I don't think the `.conf` file for a particular board is automatically picked up?

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: samples/subsys/usb/mass/CMakeLists.txt (hunk 1 above)
Comment (carlescufi) [reply samples/subsys/usb/mass/CMakeLists.txt]: My mistake, this was added a while back and I didn't know:
https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/app/boilerplate.cmake#L344

**Review by carlescufi — APPROVED** [review]

**Review by finikorg — APPROVED** [review]
LGTM

File: samples/subsys/usb/mass/prj_nrf52840_pca10056.conf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+CONFIG_STDOUT_CONSOLE=y
```
Comment (jfischer-phytec-iot) [inline samples/subsys/usb/mass/prj_nrf52840_pca10056.conf]: Is this file necessary? It would fit a lot of boards and it is not that it is tuned for pca10056. The same configuration to build is also given in `prj.conf` and `overlay-ram-disk.conf` or do I miss something?

Comment (masz-nordic) [reply samples/subsys/usb/mass/prj_nrf52840_pca10056.conf]: For now it is `proj.conf` and `overlay-ram-disk.conf` combined. After #14459 is fixed, I want to add `overlay-flash-disk.conf` too. Is there a way to define a .conf file specific to multiple boards?

Comment (finikorg) [reply samples/subsys/usb/mass/prj_nrf52840_pca10056.conf]: Do we really need to duplicate those 2 configuration files?

Comment (finikorg) [reply samples/subsys/usb/mass/prj_nrf52840_pca10056.conf]: Can you do this in cmake files? @carlescufi

**Comment by galak** [comment]
I accidentally merged this, let me know if its an issue and we can revert.

**Comment by jfischer-phytec-iot** [comment]
@galak is not so bad, @masz-nordic if you on it, can you please remove prj_nrf52840_pca10056.conf in subsequent PR that will resolve #14459?
